front/app/modules/commercial/insights/admin/containers/Insights/Edit/InputsTable/ScanCategory.tsx

```diff
@@ -69,50 +69,66 @@ type ScanCategoryProps = {
   progress: number;
   triggerScan: () => void;
   onClose: () => void;
+  cancelScan: () => void;
 } & InjectedIntlProps;
 
-const scanCategoryMessagesMap: Record<
-  ScanStatus,
-  {
-    title: MessageDescriptor;
-    description: MessageDescriptor;
-    button?: MessageDescriptor;
-  }
-> = {
-  isIdle: {
-    title: messages.categoriesScanTitle,
-    description: messages.categoriesScanDescription,
-    button: messages.categoriesScanButton,
-  },
-  isInitializingScanning: {
-    title: messages.categoriesScanInProgressTitle,
-    description: messages.categoriesScanInProgressDescription,
-  },
-  isScanning: {
-    title: messages.categoriesScanInProgressTitle,
-    description: messages.categoriesScanInProgressDescription,
-  },
-  isFinished: {
-    title: messages.categoriesScanDoneTitle,
-    description: messages.categoriesScanDoneDescription,
-    button: messages.categoriesScanDoneButton,
-  },
-  isError: {
-    title: messages.categoriesScanErrorTitle,
-    description: messages.categoriesScanErrorDescription,
-    button: messages.categoriesScanDoneButton,
-  },
-};
-
 const ScanCategory = ({
   intl: { formatMessage },
   status,
   progress,
   triggerScan,
+  cancelScan,
   onClose,
 }: ScanCategoryProps) => {
   const nlpFeatureFlag = useFeatureFlag({ name: 'insights_nlp_flow' });
 
+  const scanCategoryMessagesMap: Record<
```

Comment:
Since this object doesn't only store messages anymore, maybe this can be renamed? Not sure to what though- I was thinking of `scanCategoryMetaData` but that's a bit too vague. If you happen to have a good idea, would be nice to change, otherwise nevermind :)

Reply:
Yep, good point! I opted for `scanCategoryDetailsMap`, hope it makes it clearer...